src/main/kotlin/day16/repository/image/ImageRepositoryImpl.kt

```
@@ -0,0 +1,107 @@
+    }
+
+    private fun formatDuration(duration: Duration): String {
+        val seconds = duration.seconds
+        val absSeconds = abs(seconds)
+        val positive = String.format(
+            "%02d:%02d:%02d.%06d",
+            absSeconds / 3600,
+            absSeconds % 3600 / 60,
+            absSeconds % 60,
+            duration.nano / 1000
+        )
```

detekt (1.23.7)

[warning] String.format(
            "%02d:%02d:%02d.%06d",
            absSeconds / 3600,
            absSeconds % 3600 / 60,
            absSeconds % 60,
            duration.nano / 1000
        ) uses implicitly default locale for string formatting.

(detekt.potential-bugs.ImplicitDefaultLocale)